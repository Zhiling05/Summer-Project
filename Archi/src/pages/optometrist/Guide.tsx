import React from 'react';

function Guide() {
  return (
    <div>
      <h2>Optometrist Guide Page</h2>
    </div>
  );
}

export default Guide;
