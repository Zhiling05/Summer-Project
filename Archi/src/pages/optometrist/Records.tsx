
function Records() {
  return (
    <div>
      <h2>Optometrist Records Page</h2>
    </div>
  );
}

export default Records;
