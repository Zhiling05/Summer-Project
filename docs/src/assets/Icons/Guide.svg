<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<rect width="31.6672" height="31.6672" fill="url(#pattern0_285_69)"/>
<defs>
<pattern id="pattern0_285_69" patternContentUnits="objectBoundingBox" width="1" height="1">
<use xlink:href="#image0_285_69" transform="scale(0.0208333)"/>
</pattern>
<image id="image0_285_69" width="48" height="48" preserveAspectRatio="none" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAYAAABXAvmHAAAB3UlEQVR4AdXBoWuUUQAA8N/eHiJiUBh4Yc2Fr2kUBPHAYDQIRqtB2MATF4S7C6KyBZf8Cww2wbLguAUxWdYuLF44wbCgQVgwTHAfb2Pq+7557/ebc7JruI+bWMR5/9cQA79Ex1vAa9w1w6KjLWKEJTMuSJ3FeywpQJB6iKsKEdRFPFGQqO4aFuT7gm1MHFjETVzSsKiukmcHq9h0tNt4gSsaEtUt+ncb6GHf8TbxAetY1oCoGRtY8Wf2seLAskxBvh30/L0edmQK8q1iX2oJPfRQSe1jVaYozxdsSq1gDdGB53iMV+o28RUL/lGQZ1tqCWuIfotYQyW1JUOQZyJ1B1FqF3tSExmiPI/wyMnG6GIq9U2GoH1jdDGV6uCeDEG7xuhiKtXBCJUMQXvG6GIq1cEIlUxBO8boYirVwQiVBgTteIupVAcjVBoSnJ4ORqg0KDg9D1BpWNSOPvpOQVC4oHBRO4YYqBugr2FB4YLCBYULChe1o4++UxAULihcULigcEHhgsJFf+YZ1rGnXQP01Q0xcIzoZFt4akYFJ/tohgV1E6nrZlhQ91nqFtZwwQyak9rFZbNrziFBqqcgQeodXirEvKN9wHfcQDRbhg6Zd7xPeIMfuIhzOOP/GzrkJw1bT/kYmVI0AAAAAElFTkSuQmCC"/>
</defs>
</svg>
