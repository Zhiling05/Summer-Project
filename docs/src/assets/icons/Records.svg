<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<rect width="32" height="32" fill="url(#pattern0_285_65)"/>
<defs>
<pattern id="pattern0_285_65" patternContentUnits="objectBoundingBox" width="1" height="1">
<use xlink:href="#image0_285_65" transform="scale(0.0208333)"/>
</pattern>
<image id="image0_285_65" width="48" height="48" preserveAspectRatio="none" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAYAAABXAvmHAAAB5UlEQVR4AdXBsUtUcRwA8I9fHyJ0NDlYHdFwwxvcCoKI1iROuKUhaCuQyKHRhv6CoIYKhdYGoeXAoA5aggaH/gAFBYdwbLrhDQ4NDnFq573XO7vf5zOluhU8RW68tvEOb51iWjXreIE54zeHe7iET44J5S1g2flbxoJjQnmL/p9Fx2TKaxjUQweFes2ii7v+aDgm/LstFOpXYMsZMqdbwVPkKPAFD1CYMOGkdbxB7sgsOnhoAoVBC1iWkMygRScV6OGDCZQZ1DCohw4K1eTYRMtotrGEXSMKw22hUF0HLaPL0VZCGK8N7BndDrpKyIzXPlrGKCQuJC5Tr2v4jFw121jCrhGFenWQqy5HWwmhXl3sqG4HXSVk6rWP3DkKiQuJy9QrxyZaqtnGEnaNKNSrg5bqcrSVEOq1gT3V7aCrhEy99tFyjkJ5hwYV/qMw3HXMGvQFh44c4qvxmMVNZ5hChpd4jIa/+4kn+IQbuI3v+KFebayh6e/6WMfzabzCM8wY7iLu4DUOsIUD9fuGy4abwS1cCDwyuibmjU8T80a3HGgo5z2a6tfEmnIamfLaaJsQIXEhcSFxIXEhcSFxIXEhcSFxIXEhcYG+dPUDPenqBVbRl54+VqfxCx9xBVcxY7L1sYn72P0NS+pUu8/wi5cAAAAASUVORK5CYII="/>
</defs>
</svg>
