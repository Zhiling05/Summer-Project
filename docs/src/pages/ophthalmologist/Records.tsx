const Records = () => {
  return <div>This is the Records Page</div>;
};
export default Records;
