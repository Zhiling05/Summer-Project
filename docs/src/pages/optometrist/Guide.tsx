import { useNavigate } from 'react-router-dom';
import '../../styles/question.css';
import BottomNav from '../../components/BottomNav';

import NHSLogo  from '../../assets/NHS_LOGO.jpg';
import DIPPLogo from '../../assets/DIPP_Study_logo.png';

export default function Guide() {
  const nav = useNavigate();

  return (
    <>
      {/* 顶栏（保持通用写法） */}
      <header className="nhs-header">
        <div className="nhs-header__inner">
          <img className="logo nhs-logo" src={NHSLogo}  alt="NHS logo" />
          <img className="logo dipp-logo" src={DIPPLogo} alt="DIPP Study logo" />
          <span className="nhs-header__service">Guide</span>
        </div>
      </header>

      {/* ★★★ 与 Records 同级的“全宽灰底” ★★★ */}
      <div
        style={{
          background: 'var(--page-grey)',   // 同一灰色
          width: '100vw',
          marginLeft: 'calc(-50vw + 50%)', // 左右拉伸
          minHeight: 'calc(100vh - 64px)', // 64 px 让给 BottomNav
          boxSizing: 'border-box',
          paddingTop: 'var(--space-xl)',
        }}
      >
        {/* 中央内容栅格（960 px 封顶） */}
        <div style={{ maxWidth: 960, margin: '0 auto', padding: '0 var(--space-l)' }}>
          

          <h1 className="nhsuk-heading-l" style={{ textAlign: 'center', marginBottom: 'var(--space-l)' }}>
            Guide
          </h1>

          {/* 1. Software instruction */}
          <section className="question-box" style={{ marginBottom: 'var(--space-l)' }}>
            <h2 className="nhsuk-heading-s">Software instruction</h2>
            <ol>
              <li>context line&nbsp;1</li>
              <li>context line&nbsp;2</li>
              <li>context line&nbsp;3</li>
              <li>context line&nbsp;4</li>
            </ol>
          </section>

          {/* 2. Tutorial on Assessment Workflow */}
          <section className="question-box" style={{ marginBottom: 'var(--space-l)' }}>
            <h2 className="nhsuk-heading-s">Tutorial on Assessment Workflow</h2>
            <ol>
              <li>context line&nbsp;1</li>
              <li>context line&nbsp;2</li>
              <li>context line&nbsp;3</li>
              <li>context line&nbsp;4</li>
            </ol>
          </section>

          {/* 3. Reference Image Gallery */}
          <section className="question-box" style={{ marginBottom: 'var(--space-l)' }}>
            <h2 className="nhsuk-heading-s">Reference Image Gallery</h2>
            <div style={{ display: 'flex', gap: 'var(--space-m)', overflowX: 'auto' }}>
              {['context line 1', 'context line 2', 'context line 3'].map(txt => (
                <button key={txt}
                  style={{ flex: '0 0 140px', height: 80,
                           background: 'var(--border-grey)', border: 'none',
                           borderRadius: 'var(--radius-s)', color: 'var(--text-inverse)',
                           fontWeight: 600 }}>
                  {txt}
                </button>
              ))}
            </div>
          </section>

          {/* 4. DIPP Website Link */}
          <section className="question-box" style={{ marginBottom: 'calc(64px + var(--space-xl))' }}>
            <h2 className="nhsuk-heading-s">DIPP Website Link</h2>
            <a
              href="https://www.bristol.ac.uk/primaryhealthcare/researchthemes/dipp-study/"
              target="_blank" rel="noopener noreferrer"
              style={{
                display: 'inline-block',
                background: 'var(--page-grey)',
                padding: 'var(--space-s) var(--space-m)',
                borderRadius: 'var(--radius-s)',
                color: 'var(--core-blue)',
                fontWeight: 600,
              }}
            >
              Visit DIPP Study website ↗
            </a>
          </section>
        </div>
      </div>

      {/* 底部导航（仍然 fixed） */}
      <BottomNav />
    </>
  );
}
