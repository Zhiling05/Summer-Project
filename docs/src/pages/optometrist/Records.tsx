import { useMemo, useState, useEffect } from "react";

import { useNavigate } from "react-router-dom";

import BackButton from '../../components/BackButton';
import "react-datepicker/dist/react-datepicker.css";
import { parseISO, format } from "date-fns";
import { http } from "../../api/index";

import Header from "../../components/Header";
import BottomNav from "../../components/BottomNav";
import "../../styles/records.css";

import { AssessmentRecommendations } from "../../types/recommendation";
import Sidebar from '../../components/SideBar'; 

import DatePicker, { registerLocale } from "react-datepicker";
import { enGB } from "date-fns/locale";
import 'react-datepicker/dist/react-datepicker.css';
registerLocale('en-GB', enGB);



const LEVELS: Level[] = ["high", "medium", "low"];
export type Level = "high" | "medium" | "low";

type RiskLabel = NonNullable<AssessmentRecommendations["recommendationType"]>;

function normalizeRisk(raw: string): RiskLabel {
  return raw.toLowerCase().replace(/_/g, "-") as RiskLabel;
}

type NormalizedRecord = {
  id: string;
  date: string;
  risk: RiskLabel;
};

/** UI 配置 */
const LEVEL_UI: Record<Level, { css: "red" | "orange" | "green"; title: string; desc: string }> = {
  high:   { css: "red",    title: "High Risk",   desc: "Urgent or Immediate Referral Required" },
  medium: { css: "orange", title: "Moderate Risk", desc: "Further Observation Required" },
  low:    { css: "green",  title: "Low Risk",   desc: "No Special Intervention Required" },
};

const RISK_TO_LEVEL: Record<RiskLabel, Level> = {
  "emergency-department":   "high",
  "immediate":              "high",
  "urgent-to-oph":          "medium",
  "urgent-to-gp-or-neur":   "medium",
  "to-gp":                  "low",
  "no-referral":            "low",
  "other-eye-conditions-guidance":    "medium",
};

const formatDateForDisplay = (iso: string) => {
  const date = parseISO(iso);
  return format(date, 'dd-MM-yyyy');
};
type FilterValue = Level | "all";

const PAGE_SIZE = 20;

export default function Records() {
  const navigate = useNavigate();
  const [remoteRecs, setRemoteRecs] = useState<NormalizedRecord[]>([]);
  const [loading, setLoading] = useState(false);
  const [currentPage, setCurrentPage] = useState(1);
  const [level, setLevel] = useState<FilterValue>("all");
  const [from, setFrom] = useState<Date | null>(null);
  const [to, setTo] = useState<Date | null>(null);

  // 获取所有数据
  useEffect(() => {
    setLoading(true);
    http('/assessments')
       .then((json: any) => {
          const arr = Array.isArray(json) ? json : (json?.records ?? []);
          const normalized = arr.map((r: any) => {

          const date = r.createdAt ?? r.date;            // 统一时间
          const raw  = r.recommendation ?? r.risk;       // 统一推荐
          const risk = (normalizeRisk(String(raw)) as RiskLabel) || "no-referral";
          return { id: r.id, date, risk };
        });
        setRemoteRecs(normalized);
      })
      .catch(console.error)
      .finally(() => setLoading(false));
  }, []);

  const all: NormalizedRecord[] = remoteRecs;  

  // —— 统计（基于所有数据） ——
  const stats = useMemo(() => {
    const initial = { high: 0, medium: 0, low: 0 } as Record<Level, number>;
    return all.reduce((acc, rec) => {
      const lvl = RISK_TO_LEVEL[rec.risk];
      acc[lvl] += 1;
      return acc;
    }, initial);
  }, [all]);

  // —— 过滤（应用筛选条件） ——
  const filteredList = useMemo(() => {
    return all.filter((r) => {
      const lvl = RISK_TO_LEVEL[r.risk];
      if (level !== "all" && lvl !== level) return false;

      const recDate = parseISO(r.date);
      if (from && recDate < from) return false;
      if (to   && recDate > to)   return false;
      return true;
    });
  }, [all, level, from, to]);

  // —— 分页计算 ——
  const totalPages = Math.ceil(filteredList.length / PAGE_SIZE);
  const startIndex = (currentPage - 1) * PAGE_SIZE;
  const endIndex = startIndex + PAGE_SIZE;
  const currentPageData = filteredList.slice(startIndex, endIndex);

  // —— 分页控制 ——
  const goToPage = (page: number) => {
    if (page >= 1 && page <= totalPages) {
      setCurrentPage(page);
    }
  };

  const goToPrevPage = () => {
    if (currentPage > 1) {
      setCurrentPage(currentPage - 1);
    }
  };

  const goToNextPage = () => {
    if (currentPage < totalPages) {
      setCurrentPage(currentPage + 1);
    }
  };

  // 当筛选条件改变时，重置到第一页
  useEffect(() => {
    setCurrentPage(1);
  }, [level, from, to]);

  // 用于显示新提交记录的函数（将来实现增量更新时使用）
  const addNewRecord = (newRecord: NormalizedRecord) => {
    setRemoteRecs(prev => [newRecord, ...prev]); // 插入到顶部
    setCurrentPage(1); // 跳转到第一页显示新记录
  };

  return (
      <>
        <Header title="Records" />
        <BackButton />
        <Sidebar /> 
        
        <main className="records-main">
          <div className="records-wrapper">
            <h1 className="records-title">Records</h1>

            {/* 统计卡 */}
            <section className="stats-grid">
              {LEVELS.map((lvl) => (
                  <article key={lvl} className={`stats-card ${LEVEL_UI[lvl].css}`}>
                    <span className="stats-title">{LEVEL_UI[lvl].title}</span>
                    <span className="stats-num">{stats[lvl]}</span>
                    <span className="stats-desc">{LEVEL_UI[lvl].desc}</span>
                  </article>
              ))}
            </section>

            {/* 筛选栏 */}
            <section className="filter-bar">
              <label>
                Risk Level
                <select value={level} onChange={(e) => setLevel(e.target.value as FilterValue)}>
                  <option value="all">All</option>
                  <option value="high">High Risk</option>
                  <option value="medium">Moderate Risk</option>
                  <option value="low">Low Risk</option>
                </select>
              </label>
              <label>
                Start Date
                <DatePicker
                    selected={from}
                    onChange={(date) => setFrom(date)}
                    placeholderText="DD-MM-YY"
                    dateFormat="dd-MM-yy"
                    locale="en-GB"
                    isClearable
                    showMonthDropdown
                    showYearDropdown
                    dropdownMode="select"
                />
              </label>
              <label>
                End Date
                <DatePicker
                selected={to}
                onChange={(date) => setTo(date)}
                placeholderText="DD-MM-YY"
                dateFormat="dd-MM-yy"
                locale="en-GB"
                isClearable
                showMonthDropdown
                showYearDropdown
                dropdownMode="select"
                />
              </label>
            </section>

            {/* 分页信息 */}
            {!loading && filteredList.length > 0 && (
              <div style={{ 
                display: 'flex', 
                justifyContent: 'space-between', 
                alignItems: 'center', 
                marginBottom: 'var(--space-l)',
                fontSize: '0.9rem',
                color: 'var(--text-body)'
              }}>
                <span>
                  Showing {startIndex + 1}-{Math.min(endIndex, filteredList.length)} of {filteredList.length} records
                </span>
                <span>
                  Page {currentPage} of {totalPages}
                </span>
              </div>
            )}

            {/* 表格 */}
            <div className="table-wrapper">
              <table className="records-table">
                <thead>
                <tr>
                  <th>ID</th>
                  <th>Date</th>
                  <th>Risk Level</th>
                  <th></th>
                </tr>
                </thead>
                <tbody>
                {loading ? (
                  <tr>
                    <td colSpan={4} style={{ textAlign: "center", padding: "24px" }}>
                      Loading records...
                    </td>
                  </tr>
                ) : currentPageData.length > 0 ? (
                  currentPageData.map((r) => {
                    const lvl = RISK_TO_LEVEL[r.risk];
                    return (
                        <tr key={r.id}>
                          <td>{r.id}</td>
                          <td>{formatDateForDisplay(r.date)}</td>
                          <td>
                            <span className={`tag-pill tag-${LEVEL_UI[lvl].css}`}>{LEVEL_UI[lvl].title}</span>
                          </td>
                          <td>
                            <button
                                className="view-btn"
                                onClick={() => navigate(`/optometrist/assess/recommendations/${r.risk.toUpperCase().replace(/-/g, '_')}/${r.id}`)}
                            >
                              View Details
                            </button>
                          </td>
                        </tr>
                    );
                  })
                ) : (
                  <tr>
                    <td colSpan={4} style={{ textAlign: "center", padding: "24px" }}>
                      No Matching Records Found
                    </td>
                  </tr>
                )}
                </tbody>
              </table>
            </div>

            {/* 分页控件 */}
            {!loading && totalPages > 1 && (
              <div style={{
                display: 'flex',
                justifyContent: 'center',
                alignItems: 'center',
                gap: 'var(--space-m)',
                marginTop: 'var(--space-l)',
                padding: 'var(--space-m) 0'
              }}>
                <button
                  onClick={goToPrevPage}
                  disabled={currentPage === 1}
                  style={{
                    padding: '8px 16px',
                    background: currentPage === 1 ? 'var(--border-grey)' : 'var(--core-blue)',
                    color: currentPage === 1 ? 'var(--text-body)' : 'white',
                    border: 'none',
                    borderRadius: '4px',
                    cursor: currentPage === 1 ? 'not-allowed' : 'pointer',
                    fontSize: '0.9rem'
                  }}
                >
                  Previous
                </button>

                {/* 页码按钮 */}
                <div style={{ display: 'flex', gap: '4px' }}>
                  {Array.from({ length: totalPages }, (_, i) => i + 1)
                    .filter(page => {
                      // 显示当前页前后2页
                      return Math.abs(page - currentPage) <= 2 || page === 1 || page === totalPages;
                    })
                    .map((page, index, arr) => {
                      // 添加省略号
                      const prevPage = arr[index - 1];
                      const showEllipsis = prevPage && page - prevPage > 1;
                      
                      return (
                        <div key={page} style={{ display: 'flex', alignItems: 'center', gap: '4px' }}>
                          {showEllipsis && <span style={{ padding: '0 8px' }}>...</span>}
                          <button
                            onClick={() => goToPage(page)}
                            style={{
                              padding: '8px 12px',
                              background: page === currentPage ? 'var(--core-blue)' : 'var(--lighter-base)',
                              color: page === currentPage ? 'white' : 'var(--text-body)',
                              border: page === currentPage ? 'none' : '1px solid var(--border-grey)',
                              borderRadius: '4px',
                              cursor: 'pointer',
                              fontSize: '0.9rem',
                              minWidth: '40px'
                            }}
                          >
                            {page}
                          </button>
                        </div>
                      );
                    })}
                </div>

                <button
                  onClick={goToNextPage}
                  disabled={currentPage === totalPages}
                  style={{
                    padding: '8px 16px',
                    background: currentPage === totalPages ? 'var(--border-grey)' : 'var(--core-blue)',
                    color: currentPage === totalPages ? 'var(--text-body)' : 'white',
                    border: 'none',
                    borderRadius: '4px',
                    cursor: currentPage === totalPages ? 'not-allowed' : 'pointer',
                    fontSize: '0.9rem'
                  }}
                >
                  Next
                </button>
              </div>
            )}

          </div>
        </main>
        
        <BottomNav />
      </>
  );
}