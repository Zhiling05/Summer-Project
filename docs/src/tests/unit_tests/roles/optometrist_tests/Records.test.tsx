import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import Records from '../../../../pages/optometrist/Records';
import Guide from '../../../../pages/optometrist/guide/GuideHome';
import OptometristHome from '../../../../pages/optometrist/OptometristHome';
import AssessStart from '../../../../pages/optometrist/assess/StartPage';

describe('Records Page - TDD Flow', () => {
    const renderWithRouter = (initialPath = '/optometrist/records') => {
        render(
            <MemoryRouter initialEntries={[initialPath]}>
                <Routes>
                    <Route path="/optometrist/records" element={<Records />} />
                    <Route path="/optometrist/guide" element={<Guide />} />
                    <Route path="/optometrist/home" element={<OptometristHome />} />
                    <Route path="/optometrist/assess/start" element={<AssessStart />} />
                </Routes>
            </MemoryRouter>
        );
    };

    it('should render logos and page title', () => {
        renderWithRouter();

        expect(screen.getByAltText(/NHS logo/i)).toBeInTheDocument();
        expect(screen.getByAltText(/DIPP Study logo/i)).toBeInTheDocument();
        expect(screen.getByRole('heading', { name: /Records/i })).toBeInTheDocument();
    });

    it('should render three statistics cards as buttons (even if not interactive yet)', () => {
        renderWithRouter();
        const statsButtons = screen.getAllByRole('button');
        expect(statsButtons.length).toBeGreaterThanOrEqual(3); // Defensive test
    });

    it('should render the table and View Details buttons', () => {
        renderWithRouter();
        const table = screen.getByRole('table');
        expect(table).toBeInTheDocument();
        const detailButtons = screen.getAllByRole('button', { name: /view details/i });
        expect(detailButtons.length).toBeGreaterThan(0);
    });

    it('should navigate to correct pages via BottomNav buttons', () => {
        renderWithRouter();

        // 跳转到 Home
        fireEvent.click(screen.getByRole('button', { name: /home/i }));
        expect(screen.getByText(/Optometrist Home/i)).toBeInTheDocument();

        // 跳转到 Guide
        fireEvent.click(screen.getByRole('button', { name: /guide/i }));
        expect(screen.getByText(/Guide/i)).toBeInTheDocument();

        // 跳转到 Assess
        fireEvent.click(screen.getByRole('button', { name: /assess/i }));
        expect(screen.getByText(/Start the Assessment/i)).toBeInTheDocument();
    });

    it('should not re-navigate when clicking Records button on current Records page', () => {
        renderWithRouter();

        const recordsBtn = screen.getByRole('button', { name: /records/i });
        fireEvent.click(recordsBtn);

        // 页面仍然为 Records
        expect(screen.getByRole('heading', { name: /Optometrist Records Page/i })).toBeInTheDocument();
        expect(screen.queryByText(/Guide/i)).not.toBeInTheDocument(); // 没跳转
    });
});
