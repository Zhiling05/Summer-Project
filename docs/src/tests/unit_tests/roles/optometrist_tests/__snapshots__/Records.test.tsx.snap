// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Records matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Optometrist Records Page
    </h2>
  </div>
</DocumentFragment>
`;
